src/main/modules/ardoq/ArdoqClient.ts

```diff
@@ -1,73 +1,185 @@
+        name: componentName,
       },
       responseType: 'json',
     });
   }
 
-  private createComponent(dependencyFullName: string) {
+  private createComponent(componentName: string, workspace: ArdoqWorkspace) {
     return this.httpClient.post(
-      '/api/component',
+      '/api/v2/components',
       {
-        rootWorkspace: this.apiWorkspace,
-        name: dependencyFullName,
+        rootWorkspace: config.get(workspace),
+        name: componentName,
+        typeId: this.componentTypeLookup.get(workspace),
       },
       {
         params: {
-          workspace: this.apiWorkspace,
-          name: dependencyFullName,
+          rootWorkspace: config.get(workspace),
+          name: componentName,
+          typeId: this.componentTypeLookup.get(workspace),
         },
         responseType: 'json',
       }
     );
   }
 
-  async updateDep(d: Dependency): Promise<ArdoqComponentCreatedResponse> {
-    if (this.isCached(d)) {
-      return ArdoqComponentCreatedResponse.EXISTING;
-    }
+  private createOrGetComponent(name: string, workspace: ArdoqWorkspace): Promise<string | null> {
+    return this.searchForComponent(name, workspace).then(searchRes => {
+      if (searchRes.status === 200 && searchRes.data.values.length > 0) {
+        this.logger.debug('Found component: ' + name);
+        return searchRes.data.values[0]._id;
+      }
+      return this.createComponent(name, workspace).then(createRes => {
+        if (createRes.status !== 201) {
+          this.logger.error('Unable to create component: ' + name);
+          return null;
+        }
+        this.logger.debug('Component created: ' + name);
+        return createRes.data._id;
+      });
+    });
+  }
+
+  public createVcsHostingComponent(name: string): Promise<string | null> {
+    return this.createOrGetComponent(name, ArdoqWorkspace.ARDOQ_VCS_HOSTING_WORKSPACE);
+  }
 
-    const searchResponse = await this.searchForComponent(d.getFullName());
-    if (searchResponse.status === 200) {
-      // Can now create a relationship between the application and this object
+  public createCodeRepoComponent(name: string): Promise<string | null> {
+    return this.createOrGetComponent(name, ArdoqWorkspace.ARDOQ_CODE_REPOSITORY_WORKSPACE);
+  }
 
-      if (searchResponse.data.length !== 0) {
-        this.cacheResult(d);
-        return ArdoqComponentCreatedResponse.EXISTING;
+  private async searchForReference(
+    source: string,
+    target: string
+  ): Promise<undefined | { id: string; version?: string | undefined }> {
+    return this.httpClient
+      .get('/api/v2/references', {
+        params: {
+          source,
+          target,
+        },
+        responseType: 'json',
+      })
+      .then(res => {
+        if (res.status === 200 && res.data.values.length > 0) {
+          return {
+            id: res.data.values[0]._id,
+            version: res.data.values[0].customFields?.version,
+          };
+        }
```

Comment:
not sure I like the implicit void return here. It might be deliberate but it kind of looks like a mistake. Maybe add `else { return undefined; }` whatever